### PR TITLE
PowerOff Detector

### PR DESCRIPTION
**Pull request by sudip70**
Power off detector with dialog

### CODE REVIEW

**Comment by sudip70** [comment]
It has both frames and trigger in trigger mode and power of detector update

**Comment by sudip70** [comment]
there is some issue with logic. so I will push that and we can merge

File: python/app/g2panel [hunk 1]
```diff
@@ -345,14 +344,26 @@ class MainWindow(QtWidgets.QMainWindow):
...
+            self.rcv.frames = self.frameSpinBox.value()
+            self.det.frames = self.frameSpinBox.value()
 
     #Setting initial trigger
     def setTrigger(self):
+        
+        #If trigger mode is selected then the frame number is replaced by the triggers other wise it takes the frame number
+        if self.det.timing==timingMode.TRIGGER_EXPOSURE:
+            self.rcv.frames = self.triggerSpinBox.value()
+            self.det.frames = self.triggerSpinBox.value()
```
Comment (erikfrojdh) [inline python/app/g2panel:361]: No need to set frames when number of triggers change.

File: python/app/g2panel [hunk 2]
```diff
@@ -418,15 +429,25 @@ class MainWindow(QtWidgets.QMainWindow):
     def triggerMode(self, selected):
```
Comment (erikfrojdh) [inline python/app/g2panel:429]: I think here we also need to update the receiver for the number of frames. Scenario is that the user switches between trigger and auto without editing the number of frames.

File: python/app/g2panel [hunk 3]
```diff
@@ -418,15 +429,25 @@ class MainWindow(QtWidgets.QMainWindow):
...
-        self.det.highvoltage = 0
-        self.highVoltageSpinBox.setValue(self.det.highvoltage)
-        time.sleep(1)
-        #TODO yet to implement with the initialization of the chip
-        #self.det.powerchip = False
-
+        #Asking user if they really want to power off the detectors
+        powerDlg = QtWidgets.QMessageBox()
+        powerDlg.setWindowTitle("Power Off the Detecor!!!")
+        powerDlg.setText("Do you really want to power off the Detectors?")
```
Comment (erikfrojdh) [inline python/app/g2panel:439]: Most likely it will be a single detector so change to: Do you really want to power off the detector?

**Review by erikfrojdh — CHANGES_REQUESTED** [review]
Power off sequence looks good but we need to update how the frames are set in the receiver that we always expect the correct number.

File: python/app/g2panel (hunk 3 above)
Comment (sudip70) [reply python/app/g2panel:439]: It must be typo but I made the changes

File: python/app/g2panel (hunk 2 above)
Comment (sudip70) [reply python/app/g2panel:429]: I have added a change in frames for both auto and trigger mode

File: python/app/g2panel (hunk 1 above)
Comment (sudip70) [reply python/app/g2panel]: Removed frames

**Review by erikfrojdh — APPROVED** [review]

**Pull request merged**